ddprofiler/app/src/main/java/ddprofiler/preanalysis/PreAnalyzer.java

```diff
@@ -31,6 +32,20 @@ public class PreAnalyzer implements PreAnalysis, IO {
     private boolean knownDataTypes = false;
     private ProfilerConfig pc;
 
+    private static final HashMap<String, Pattern[]> TEMPORAL_PATTERNS = new HashMap<>() {
+        {
+            put("dateTime",
```

Comment:
- dateTime can also have a day granularity. For example 2023-02-27 00:00:00 is a dateTime, but it has the day granularity. More concretely, I think dateTime is a type and does not indicate any granularity information. In this sense, dateTime is equivalent to "Temporal".
- For temporal attributes, Let's focus on detecting things with a dateTime format and do not consider things like "monday/tuesday" for now. 
- There are many string representations for the dateTime format. You may need to add more regex expressions here.
- I'd suggest keeping these reference regex expressions in a separate file instead of in the codebase since we may need to update this file frequently and enable users to add their own expressions as well.

Reply:
- How small of a granularity would we want to keep track of? For example, would 2023-02-27 10:00:00 have hour granularity, 00:10:00 have minute granularity, and 00:00:10 have second granularity? 
- There is a chance that the first entry of a column be 2023-02-27 00:00:00, and the second entry be 2023-02-27 10:00:00. The design right now is to simply take the first match. Do we need to count matches (just like the typeOfValue method) and choose the one with most matches?
- Reference regex has been moved to a separate file here 7ebf5cc88195e9a37f9c5a734a50aebe61fa09eb

Reply:
1. As of now, I think keeping the smallest granularity as `second` is sufficient
2. This is a great point. Simply taking the first match would be a brittle solution. I like the idea of looking at the majority match.

Reply:
1. Granularity support up to second added at [83e6200](https://github.com/TheDataStation/ver/pull/71/commits/83e62009d89bf86205d1c86ca83409672420e7b1)
2. Majority match implemented at [79c968b](https://github.com/TheDataStation/ver/pull/71/commits/79c968b4ad787327db1be70f5c024873dc41c557)